### ISSUE TITLE
REST response in plugin not behaving as expected with rxjs observables

### ISSUE THREAD

**Issue by Tomibusch1** [vendure-ecommerce/vendure#1854]
**Bug Description**
According to the documentation in the following page: [URL](https://www.vendure.io/docs/typescript-api/plugin/vendure-plugin-metadata/)
"any Nestjs Module is a valid Vendure plugin".
However, if a REST route in a controller returns an observable like in the following example:

    @Get('observable')
    TestGetObservable(): Observable<any>{
        //console.log(of('testObservable'));
        return of('testObservable');
    }

The observable seems like it gets parsed as JSON and then returned. which results in an empty object ("{  }") in this simple case. This is not what would be expected if the plugin behaved like a Nest JS module. 

**To Reproduce**

1.Starting from a clean new project generated with npx @vendure/create my-app. Create this two files in an empty directory: "test.module.ts" and "test.controller.ts"
2.In the controller file ("test.controller.ts") copy the following code:

    import { Controller, Get, Post } from '@nestjs/common';
    import { of, Observable } from 'rxjs';
    
    @Controller('test-plugin')
    export class TestController {
        constructor() { }
    
          @Get('observable')
          TestGetObservable(): Observable<any>{
              //console.log(of('testObservable'));
              return of('testObservable');
          }
    }

3.In the plugin module file ("test.module.ts") copy the following code:

    import { VendurePlugin } from '@vendure/core';
    import { TestController } from './test.controller';
    
    @VendurePlugin({
        imports: [
        ],
        providers: [],
        controllers: [TestController],
    })
    export class TestModule { }

4. Add TestModule to the "plugins" array in "vendure-config.ts".

5.execute with yarn dev (or npm run dev)

6.Navigate to localhost:3000/test-plugin/observable with any browser or http client, you should get "{}" as the return.

7.Optional: If you add any "pipe" operation to the observable you will see that the return changes to {"source":{}} this leads me to believe that the observable is being parsed as JSON. Example function:

    @Get('mapped-observable')
    TestPostObservable(): Observable<any> {
        console.log(of('testObservable').pipe(map(data => data)));
        return of('testObservable').pipe( map(data => data));
    }

8. Optional: If you wish to verify what the correct return value should be, you can copy both files into a vanilla Nest JS project and swap @VendurePlugin with @Module.

**Expected behavior**
Since Nest JS automatically subscribes to observables when returned from a REST request ([URL](https://docs.nestjs.com/controllers#asynchronicity)), it would be expected for a vendure plugin to return the value of the observable after it has been subscribed to in a REST request just like Nest JS.

**Environment:**
 - @vendure/core version: 1.8.1
 - Nodejs version: v16.16.0
 - Database: postgres 14

**Current Workaround**
Currently it is possible to still use rxjs observables by manually turning them into promises with firstValueFrom(yourObservable).

**Comment by michaelbromley** [comment]
Thanks for the report!
I'd guess that something on the Vendure middleware stack is not dealing with observables correctly.

**Comment by michaelbromley** [comment]
Hi,
I attempted to reproduce this issue but cannot. I followed the instructions you gave on a fresh install of Vendure v1.8.1 via `@vendure/create`

Making a GET request to `http://localhost:3000/test-plugin/observable` results in the response:

```
testObservable
```

Furthermore, adding a map like this:
```ts
    @Get('observable')
    TestGetObservable(): Observable<any> {
        return of('testObservable').pipe(map(data => data + ' mapped'));
    }
```

results in the expected response:
```
testObservable mapped
```

I'm on Windows 10, using Node v16.15.

Is there any other changes you made to a clean install that may be worth noting?
